docs/matrix_format_spec.md

```diff
@@ -0,0 +1,101 @@
+# Format
+
+The secondary analysis pipelines shall write expression matrices and associated
+tabular metadata using the zarr version 2 directory store format. The format is
+specified more precisely
+[here](https://zarr.readthedocs.io/en/stable/spec/v2.html), but in general zarr
+stores contain groups and arrays. Groups can contain other groups and arrays.
+Arrays are stored chunked and compressed. The file names and directory
+structure of the zarr store convey the group and chunk structure.
+
+# Groups and Arrays
+
+Expression values and metadata are stored in seven arrays. These arrays are
+stored within a single zarr group called `expression_matrix`.
+
+- expression - The expression values themselves. Rows correspond to cells and
+  columns to genes. These must represent raw counts. Any normalization is
+  deferred, but metadata that enables normalization should be included in the
+  metadata arrays defined below.
+  - shape: `(n_cells, n_genes)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(10000, n_genes)`
+  - dtype: `int32`
```

Comment:
`uint32` for raw counts.

Reply:
So it turns out this actually needs to be a `float32`. For some experiment types, the "raw count" is a sum of probabilities.